tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -74,10 +88,16 @@ spec:
           repository=$(jq -r '.repository' <<< $component)
           name=$(jq -r '.name' <<< $component)
           git_sha=$(jq -r '.source.git.revision' <<< $component) # this sets the value to "null" if it doesn't exist
-          # take tag if present in mapping file, otherwise take ReleaseStrategy default provided as a task param
-          tag=$(params.tag)
-          if [ $(jq 'has("tag")' <<< $component) == "true" ] ; then
-              tag=$(jq -r '.tag' <<< $component)
+          # prefixedTag should be used when tagPrefix parameter length is non-zero
```

Comment:
Let's add a period at the end, because there is a new sentense on the next line.

Reply:
I rewrote the text. Please check if now it reads better.

Reply:
I don't like the verb "should" - the comment describes what actually happens, not what should happen ;)

Here's my shot at another version:
```
The tag is determined as follows:
If the task parameter `tagPrefix` has non-zero length, use that one (using <prefix>-<timestamp> format).
Otherwise use the tag set in the mapped snapshot for the given component if set. Otherwise use the task parameter `tag`.
```

And then, I would also change the logic to make it cleaner - using if-else:
```sh          
          if [ -n "${prefixedTag}" ] ; then
              tag="$prefixedTag"
          else
              tag=$(jq -r '.tag // "$(params.tag)"' <<< $component)
          fi
```

Reply:
Now it looks good.